java/compaction/compaction-job-execution/src/main/java/sleeper/compaction/job/execution/CompactionJobMessageHandler.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.compaction.job.execution;
+
+import com.amazonaws.services.sqs.model.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.compaction.job.CompactionJob;
+import sleeper.configuration.properties.instance.InstanceProperties;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static sleeper.configuration.properties.instance.CompactionProperty.COMPACTION_TASK_MAX_CONSECUTIVE_FAILURES;
+import static sleeper.configuration.properties.instance.CompactionProperty.COMPACTION_TASK_MAX_TIME_IN_SECONDS;
+
+public class CompactionJobMessageHandler {
```

Comment:
Maybe a better name for this would be something like CompactionTask?

We could split out separate issues to:
- Align this with IngestTask and improve one or both of the structures of how the components interact, to clarify the flow of control.
- Potentially move the handling of the CompactionTaskStatusStore into this class, to create one set of tests which describe the high level of what a compaction task does.

Reply:
I've split out these issues:
- https://github.com/gchq/sleeper/issues/1900
- https://github.com/gchq/sleeper/issues/1901